### PR TITLE
[Cmake] relocate buildtools find modules, generate_patchcommand, build flatc internal

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Move "build tool" find modules to their own module path
Move "build tool" items from required/option_deps and put into their own build tool list

implement generate_patchommand, and move existing find modules to use it
extend more modulehelper variables for BUILD_DEP_TARGET

implement a native toolchain. This is a stripped down toolchain for use with targets added via externalproject_add (ie BUILD_DEP_TARGET usage of buildhost compatible targets)

split flatbuffers find modules into 2. One for the native build tool flatc, the other for the header install

Build flatc for all non linux/freebsd platforms by default

## Motivation and context
Was really trying to avoid build tools for now, but here we are.

This is the first instance of creating a build tool as part of the core cmake process.

@wsnipex let me know what you think about the relocation of the build tool find modules. Its probably not necessary, but i thought it might make things a bit more organised.
This was prompted from your comment about the ENABLE_INTERNAL_LZO2 PR and the texturepacker failure. This would set the first bit of groundwork to enable building not only buildtools, but also their lib dependencies. Still some legwork to get that in a good state, but this is a start.
Googletest and jsonschemabuilder should be able to be done in the same manner quite easily.
swig has a dependency on pcre, so i'll need to sort out the native lib building for that and texturepacker.

Also note, the use of core_optional_dep/core_required_dep probably isnt exactly correct (function appends to some lists that build tools shouldnt really be in). Havent seen any ill effects, and didnt want to duplicate a function to do essentially the same thing that exists. Could possibly change this, but for now, it works.

## How has this been tested?
ios/macos aarch64
Jenkins test jobs for all platforms - https://jenkins.kodi.tv/job/BuildMulti-Choice/819/

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by wsnipex** [comment]
as you said, moving the find modules isn't necessary,  but it does give some structure. I don't mind either way.
We probably don't see the whole picture until native lib building is in place.

**Review by garbear — APPROVED** [review]
Seems like a natural progression. Flatc split looks good.

File: cmake/modules/buildtools/FindFlatC.cmake [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+if(ENABLE_INTERNAL_FLATBUFFERS)
+  include(cmake/scripts/common/ModuleHelpers.cmake)
+
+  set(MODULE_LC flatbuffers)
+
+  SETUP_BUILD_VARS()
+
+  # Check for existing FLATC.
+  find_program(FLATBUFFERS_FLATC_EXECUTABLE NAMES flatc)
```
[the remark this answers is not in the document]
Comment (fuzzard) [reply cmake/modules/buildtools/FindFlatC.cmake]: Im not totally sure about this in the context of a build tool.
im leaning towards removing the outer if check for ENABLE_INTERNAL_FLATBUFFERS. my reasoning is whether its provided by system or not, we dont really care that much in regards to a build tool as long as flatc is available.

Id more lean to doing a custom_command version check if anything, but im hesitant to do that right now as im not sure theres an immediate need

Ive added a fixup comit with the removal of enable_internal_flatbuffers for flatc, let me know what you think

**Review by wsnipex — APPROVED** [review]

**Pull request merged**